core/src/main/java/io/questdb/griffin/engine/functions/BinaryFunction.java

```diff
@@ -56,4 +56,8 @@ default void toTop() {
     Function getLeft();
 
     Function getRight();
+
+    default boolean isRuntimeConstant() {
+        return (getLeft().isRuntimeConstant() || getLeft().isConstant()) && (getRight().isConstant() || getRight().isRuntimeConstant());
```

Comment:
i don't feel this is correct, perhaps this:

```java
        if ((!getLeft().isConstant() && !getLeft().isRuntimeConstant()) || (!getRight().isConstant() && !getRight().isRuntimeConstant())) {
            return false;
        }
        return getLeft().isRuntimeConstant() || getRight().isRuntimeConstant();

```

collapsible to 

```java
        final Function l = getLeft();
        final Function r = getRight();
        return (l.isConstant() || l.isRuntimeConstant()) && (l.isConstant() || l.isRuntimeConstant()) && (l.isRuntimeConstant() || r.isRuntimeConstant());
```

Reply:
constant + runtime constant = runtime constant
runtime constant  + runtime constant = runtime constant
I refactored it slightly as you suggested